File: ReceptsBook.Web/Modules/ServerTypes/CookingBookDb/RecipesForm.ts
import { StringEditor, DateEditor, TextAreaEditor, ImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface RecipesForm {
    Title: StringEditor;
    CreatedDate: DateEditor;
    Description: TextAreaEditor;
    Ingredients: TextAreaEditor;
    Сooking: TextAreaEditor;
    Recommendations: StringEditor;
    PrimaryImage: ImageUploadEditor;
}

export class RecipesForm extends PrefixedContext {
    static formKey = 'CookingBookDb.Recipes';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!RecipesForm.init)  {
            RecipesForm.init = true;

            var w0 = StringEditor;
            var w1 = DateEditor;
            var w2 = TextAreaEditor;
            var w3 = ImageUploadEditor;

            initFormType(RecipesForm, [
                'Title', w0,
                'CreatedDate', w1,
                'Description', w2,
                'Ingredients', w2,
                'Сooking', w2,
                'Recommendations', w0,
                'PrimaryImage', w3
            ]);
        }
    }
}

File: ReceptsBook.Web/Modules/CookingBookDb/Recipes/RecipesForm.cs
using Serenity.ComponentModel;
using System;

namespace ReceptsBook.CookingBookDb.Forms;

[FormScript("CookingBookDb.Recipes")]
[BasedOnRow(typeof(RecipesRow), CheckNames = true)]
public class RecipesForm
{
    public string Title { get; set; }
    public DateTime CreatedDate { get; set; }

    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    [TextAreaEditor(Rows = 8)]
    public string Ingredients { get; set; }

    [TextAreaEditor(Rows = 10)]
    public string Сooking { get; set; }

    public string Recommendations { get; set; }
    public string PrimaryImage { get; set; }
    //public string GalleryImages { get; set; }
}